src/shared/directives/conditional-show/domain/conditional-show.model.ts

```diff
@@ -67,6 +68,54 @@ export class ShowCondition {
     }
     return this.matchAndConditions(fields, this.condition, path);
   }
+
+  /**
+   * Determine whether this is affected by fields that have a display_context
+   * of HIDDEN or READONLY, which means they aren't able to be changed by the
+   * user's actions.
+   *
+   * @param caseFields Inspected to see appropriate display_context(s).
+   */
+  public hiddenCannotChange(caseFields: CaseField[]): boolean {
```

Comment:
I understand lots of logic in this method hiddenCannotChange.Can we have as unit test to it?

Reply:
Absolutely! Just added 23 unit tests to cover all the scenarios I could think of - happy and unhappy paths.